python/lsst/pipe/tasks/fit_coadd_multiband.py

```diff
@@ -323,14 +359,33 @@ def build_catexps(self, butlerQC, inputRefs, inputs) -> list[CatalogExposureInpu
                 removeScarletData=True,
                 updateFluxColumns=False,
             )
-            catexps[dataId['band']] = CatalogExposureInputs(
+            catexp_dict[dataId['band']] = CatalogExposureInputs(
                 catalog=catalog,
                 exposure=exposure,
                 table_psf_fits=models_psf[dataId] if has_psf_models else astropy.table.Table(),
                 dataId=dataId,
                 id_tract_patch=id_tp,
             )
-        catexps = [catexps[band] for band in self.config.get_band_sets()[0]]
+        # This shouldn't happen unless this is called with no inputs, but check anyway
+        if dataId is None:
+            raise RuntimeError(f"Did not build any catexps for {inputRefs=}")
+        catexps = []
+        for band in self.config.get_band_sets()[0]:
+            if band in catexp_dict:
+                catexp = catexp_dict[band]
+            else:
+                # Make a dummy catexp with a dataId if there's no data
+                # This should be handled by any subtasks
+                dataId_band = dataId.to_simple(minimal=True)
+                dataId_band.dataId["band"] = band
+                catexp = CatalogExposureInputs(
+                    catalog=afwTable.SourceCatalog(),
+                    exposure=None,
+                    table_psf_fits=astropy.table.Table(),
+                    dataId=dataId.from_simple(dataId_band, universe=dataId.universe),
+                    id_tract_patch=id_tp,
+                )
```

Comment:
I wonder if it would be useful to have an `EmptyExposureInputs` class that inherits from `CatalogExposureInputs` but only takes a dataId and id_tract_patch. This would make it clear here that this is just a placeholder.

Reply:
Hmm, it's possible. I was also consider just implementing an `is_dummy`/`is_placeholder` method that just returns `len(self.catalog) == 0`. I decided to defer committing to a particular interface until I have a clearer understanding of the benefits/drawbacks of dummy data.